File: backend/src/main/java/com/smartsolutions/hub/service/SiteService.java
package com.smartsolutions.hub.service;

import com.smartsolutions.hub.exception.ResourceNotFoundException;
import com.smartsolutions.hub.model.Site;
import com.smartsolutions.hub.repository.SiteRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Site Service
 * Handles business logic for construction sites
 */
@Service
@RequiredArgsConstructor
@Slf4j
public class SiteService {
    
    private final SiteRepository siteRepository;
    
    /**
     * Get all sites
     */
    @Transactional(readOnly = true)
    public List<Site> getAllSites() {
        log.debug("Getting all sites");
        return siteRepository.findAll();
    }
    
    /**
     * Get all active sites
     */
    @Transactional(readOnly = true)
    public List<Site> getActiveSites() {
        log.debug("Getting all active sites");
        return siteRepository.findByStatus("active");
    }
    
    /**
     * Get site by ID
     */
    @Transactional(readOnly = true)
    public Site getSiteById(Long id) {
        log.debug("Getting site by id: {}", id);
        return siteRepository.findById(id)
            .orElseThrow(() -> new ResourceNotFoundException("Site not found with id: " + id));
    }
    
    /**
     * Get sites by IDs (for user's assigned sites)
     */
    @Transactional(readOnly = true)
    public List<Site> getSitesByIds(List<Long> ids) {
        log.debug("Getting sites by ids: {}", ids);
        if (ids == null || ids.isEmpty()) {
            return List.of();
        }
        return siteRepository.findByIdIn(ids);
    }
    
    /**
     * Get sites by location
     */
    @Transactional(readOnly = true)
    public List<Site> getSitesByLocation(String location) {
        log.debug("Getting sites by location: {}", location);
        return siteRepository.findByLocation(location);
    }
}

File: frontend/src/services/siteService.ts
import apiClient from './apiClient';

export interface Site {
  id: number;
  name: string;
  location: string;
  status: string;
  createdAt?: string;
  updatedAt?: string;
}

export interface SiteDTO {
  id: number;
  name: string;
  location: string;
  status: string;
}

class SiteService {
  /**
   * Get current user's assigned sites (legacy endpoint - returns DTOs)
   */
  async getUserSites(): Promise<SiteDTO[]> {
    const response = await apiClient.get<SiteDTO[]>('/sites/me');
    return response.data;
  }

  /**
   * Get all sites
   */
  async getAllSites(): Promise<Site[]> {
    const response = await apiClient.get<Site[]>('/sites');
    return response.data;
  }

  /**
   * Get active sites only
   */
  async getActiveSites(): Promise<Site[]> {
    const response = await apiClient.get<Site[]>('/sites?status=active');
    return response.data;
  }

  /**
   * Get site by ID
   */
  async getSiteById(id: number): Promise<Site> {
    const response = await apiClient.get<Site>(`/sites/${id}`);
    return response.data;
  }

  /**
   * Get sites by IDs
   */
  async getSitesByIds(ids: number[]): Promise<Site[]> {
    const response = await apiClient.post<Site[]>('/sites/by-ids', ids);
    return response.data;
  }

  /**
   * Get sites by location
   */
  async getSitesByLocation(location: string): Promise<Site[]> {
    const response = await apiClient.get<Site[]>(`/sites/location/${location}`);
    return response.data;
  }
}

export default new SiteService();
